cakm/ekm/cakm_ekm_db_restoration.sql

```diff
@@ -0,0 +1,154 @@
+DECLARE @sql NVARCHAR(MAX);
+DECLARE @login_cmd NVARCHAR(MAX);
+DECLARE @cred_cmd NVARCHAR(MAX);
+DECLARE @alter_login NVARCHAR(MAX);
+DECLARE @dbname NVARCHAR(100);
+DECLARE @backupp VARCHAR(200);
+DECLARE @drop_sql NVARCHAR(MAX);
+DECLARE @cm_user NVARCHAR(128);
+DECLARE @cm_passwd NVARCHAR(128);
+DECLARE @login NVARCHAR(128);
+DECLARE @cred NVARCHAR(128);
+DECLARE @prop_file NVARCHAR(MAX);
+DECLARE @prop_file_bkp NVARCHAR(MAX);
+DECLARE @prop_path NVARCHAR(MAX);
+DECLARE @cakm_prop_change NVARCHAR(MAX);
+DECLARE @exec_prop_bkp_cmd NVARCHAR(MAX);
+DECLARE @exec_prop_cmd NVARCHAR(MAX);
+DECLARE @exec_exec_uuid_cmd NVARCHAR(MAX);
+DECLARE @exec_exec_fp_cmd NVARCHAR(MAX);
+DECLARE @exec_rm_prop_bkp_cmd NVARCHAR(MAX);
+
+SET @keyname = N'$(keyname)';
+SET @provider = N'$(provider)';
+SET @dbname = N'$(dbname)';
+--SET @backupp = N'$(backupp)';
+SET @backupp = N'$(backup)';
+SET @cm_user = N'$(cm_user)';
+SET @cm_passwd = N'$(cm_passwd)';
+SET @login = N'login_' + @keyname;
+SET @cred = N'cred_' + @keyname;
+SET @prop_path =N'$(prop_path)';
+ 
+IF '$(prop_path)' = ''
+BEGIN
+    SET @prop_file =N'C:\Program Files\CipherTrust\CAKM For SQLServerEKM\cakm_mssql_ekm.properties';
+	SET @prop_file_bkp =N'C:\Program Files\CipherTrust\CAKM For SQLServerEKM\cakm_mssql_ekm.properties_bkp';
+	SET @cakm_prop_change =N'C:\Program Files\CipherTrust\CAKM For SQLServerEKM\cakm_ekm_config_change.exe';
+	PRINT 'Property file path not provided, So using default path';
+END
+ELSE
+BEGIN
+	SET @prop_file = @prop_path + N'\cakm_mssql_ekm.properties';
+	SET @prop_file_bkp = @prop_path + N'\cakm_mssql_ekm.properties_bkp';
+	SET @cakm_prop_change = @prop_path + N'\cakm_ekm_config_change.exe';
+END
+
+SET @exec_prop_bkp_cmd = 'EXEC xp_cmdshell ''cp "' + @prop_file + '" "' + @prop_file_bkp + '"''';
+SET @exec_prop_cmd = 'EXEC xp_cmdshell ''cp "' + @prop_file_bkp + '" "' + @prop_file + '"''';
+SET @exec_rm_prop_bkp_cmd = 'EXEC xp_cmdshell ''rm "' + @prop_file_bkp + '"''';
+
+SET @exec_exec_uuid_cmd = 'EXEC xp_cmdshell ''"' + @cakm_prop_change + '" UUID'';';
+SET @exec_exec_fp_cmd = 'EXEC xp_cmdshell ''"' + @cakm_prop_change + '" FP'';';
+
+-- Execute the command
+PRINT "Creating the Backup of Property file";
+--PRINT @exec_prop_bkp_cmd;
+EXEC sp_executesql @exec_prop_bkp_cmd;
+
+PRINT "Setting the VKM_mode to 'yes' in Property file"
+--PRINT @exec_exec_uuid_cmd;
+EXEC sp_executesql @exec_exec_uuid_cmd;
+
+--PRINT @prop_file;
+--PRINT @prop_file_bkp;
+--PRINT @cakm_prop_change;
+
+PRINT "Getting Asymmetric Key from Key Manager"
+
+SET @sql = '
+CREATE ASYMMETRIC KEY ' + QUOTENAME(@keyname) + '
+FROM PROVIDER ' + QUOTENAME(@provider) + '
+WITH 
+PROVIDER_KEY_NAME = ''' + @keyname + ''',
+CREATION_DISPOSITION = OPEN_EXISTING;';
+
+--PRINT @sql;
+EXEC sp_executesql @sql;
+--SELECT name,thumbprint FROM SYS.ASYMMETRIC_KEYS where name = @keyname;
+
+PRINT "Setting login and cred for the key"
+
+SET @login_cmd = '
+CREATE LOGIN ' + QUOTENAME(@login) + '
```

Comment:
What if these queries fail ?

Reply:
error condition would be displayed on screen but script will continue.